airbyte-webapp/src/components/connectorBuilder/StreamTestingPanel/StreamTestButton.tsx

```diff
@@ -39,7 +49,7 @@ export const StreamTestButton: React.FC<StreamTestButtonProps> = ({ readStream }
     tooltipContent = <FormattedMessage id="connectorBuilder.invalidYamlTest" />;
   }
 
-  if (editorView === "ui" && hasErrors(true, ["global", testStreamIndex])) {
+  if (editorView === "ui" && (hasErrors(true, ["global", testStreamIndex]) || hasConfigJsonErrors)) {
```

Comment:
```suggestion
  if ((editorView === "ui" && hasErrors(true, ["global", testStreamIndex]) || hasConfigJsonErrors)) {
```
I think the parenthesis needs to be to the left of `editorView === "ui"`, otherwise the warning icon and tooltip are not shown in this case in the yaml view